zapisy/apps/grade/poll/utils.py

```diff
@@ -84,12 +87,23 @@ def group(entries: List[Poll], sort=False) -> dict:
 
     for entry in entries:
         if entry is not None:
+            is_own = employee in [entry.owner, entry.teacher, entry.gcowner]
             category = entry.category
             subcategory = entry.subcategory
+            href = reverse('grade-poll-results',
+                           kwargs={'semester_id': semester_id, 'poll_id': entry.id})
             if subcategory not in grouped_entries[category]:
                 if entry.semester:  # whether the entry is a general poll
-                    output[category].append(entry)
-                grouped_entries[category].append(entry)
+                    output[category].append(entry.to_dict())
+                    output[category].append({
+                        'number_of_submissions': entry.number_of_submissions,
+                        'is_own': is_own,
+                        'href': href})
+                grouped_entries[category].append(entry.to_dict())
+                grouped_entries[category].append({
+                        'number_of_submissions': entry.number_of_submissions,
+                        'is_own': is_own,
+                        'href': href})
```

Comment:
Bezpośrednio po wyliczeniu zmiennej `href` zrobiłbym
```
entrydict = entry.to_dict()
entrydict.update({
    'number_of_submissions': entry.number_of_submissions,
    'is_own': is_own,
    'href': href})
```
a tu skorzystałbym z niego tak:
```suggestion
                    output[category].append(entrydict)
                grouped_entries[category].append(entrydict)
```
W szczególności, obecnie coś mocno nie działa jeśli chodzi o rozwijanie kategorii w liście ankiet (i możliwe że również przy przechodzeniu do widoku innych ankiet, ale to tylko zgadywanka, skoro nic się nie rozwija…) i mam mocne podejrzenia że przynajmniej w części wiąże się to z faktem, że obecnie mocno zmieniła się zawartość generowanej tutaj struktury, w szczególności na listach zawartych w słowniku `grouped_entries` czy `output` mamy teraz dwa razy więcej elementów (które same są słownikami) niż kiedyś.